docs/sql-data-sources-jdbc.md

```diff
@@ -202,13 +202,27 @@ the following case-insensitive options:
     <td><code>keytab</code></td>
     <td>
      Location of the kerberos keytab file (which must be pre-uploaded to all nodes either by <code>--files</code> option of spark-submit or manually) for the JDBC client. When path information found then Spark considers the keytab distributed manually, otherwise <code>--files</code> assumed. If both <code>keytab</code> and <code>principal</code> are defined then Spark tries to do kerberos authentication.
+     It worth to mention that kerberos authentication with keytab is not always supported by the JDBC driver.<br><br>
+     Before using this feature please make sure the following requirements met:
+      <ol>
+         <li> The included JDBC driver version supports kerberos authentication with keytab. </li>
+         <li> There is an embedded connection provider which supports the used database (many database support already added, but since it's changing please double check the list). </li> 
+      </ol>
+     If any other kind of custom authentication is needed <code>JdbcConnectionProvider</code> developer API is added which allows developers to implement it.
```

Comment:
Could you add a link to `JdbcConnectionProvider.md`?

Reply:
Until now I haven't seen that developer documentation goes into the normal user docs.
I was following the delegation token readme file can be found [here](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/security/README.md).
Of course this doesn't mean I'm not willing to do that, just would like to ask back whether this is the way we would like to proceed. Waiting on your opinion.

Reply:
Ah, ok. It seems we don't need to do so.

Reply:
Thanks for sharing your opinion, resolving.